conjure-java-core/src/main/java/com/palantir/conjure/java/services/UndertowServiceHandlerGenerator.java

```diff
@@ -405,6 +408,9 @@ private TypeName immutableCollection(TypeName input) {
             } else if (SET_NAME.equals(parameterized.rawType)) {
                 return ParameterizedTypeName.get(
                         IMMUTABLE_SET_NAME, parameterized.typeArguments.toArray(new TypeName[0]));
+            } else if (MAP_NAME.equals(parameterized.rawType)) {
+                return ParameterizedTypeName.get(
+                        IMMUTABLE_MAP_NAME, parameterized.typeArguments.toArray(new TypeName[0]));
```

Comment:
Does the concern from https://github.com/palantir/conjure-java/pull/1051#discussion_r473081559 still apply? If you've opted in to `nonNullCollections` it doesn't feel unreasonable to disallow null keys.

I think this is fairly unlikely to break people in practice.

Reply:
Ack, will revert this.

It would be nice if there were some way for consumers to opt-in to this.